middlewares/authzCourseInstance.sql

```diff
@@ -1,6 +1,6 @@
 -- BLOCK select_authz_data
 SELECT
-    coalesce($force_mode, ip_to_mode($ip, $req_date)) AS mode,
+    coalesce($force_mode, ip_to_mode($ip, $req_date, $authn_user_id)) AS mode,
```

Comment:
Should this be `authn_user_id` or just `user_id`?

Reply:
`authn_user_id` is, indeed, what's passed to `select_authz_data` (with the "old permissions scheme") so this looks correct to me - overrides are handled with the separate SQL block `select_effective_authz_data`

Reply:
What if an instructor chooses to make the effective user a student with an active reservation? Should the mode be forced to `Exam` as it would if the student were the authn user? (Right now, it won't be, because no changes have been made to the SQL block `select_effective_authz_data`.)

Reply:
Because instructors can manually force emulated modes, I don't think we need to trigger mode based on reservation when an instructor emulates a student. Is that reasonable?

Reply:
Seems fine, at least for this first version.